core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -294,13 +300,16 @@ private[spark] class Executor(
     threadPool.shutdown()
 
     // Notify plugins that executor is shutting down so they can terminate cleanly
-    Utils.withContextClassLoader(replClassLoader) {
-      executorPlugins.foreach { plugin =>
-        try {
-          plugin.shutdown()
-        } catch {
-          case e: Exception =>
-            logWarning("Plugin " + plugin.getClass().getCanonicalName() + " shutdown failed", e)
+    if (!executorShutdown) {
```

Comment:
This is too late to check ; you'd potentially execute some of stop() twice

Reply:
@srowen I moved the check variable to the beginning of the method. Please review the changes.